src/rabbit_core_metrics.erl

```diff
@@ -384,3 +389,34 @@ get_gen_server2_stats(Pid) ->
         [] ->
             not_found
     end.
+
+auth_attempt_succeeded(Ip, Username) ->
+    %% {Id, Total, Succeeded, Failed}
+    update_auth_attempt(Ip, Username, [{2, 1}, {3, 1}]).
+
+auth_attempt_failed(Ip, Username) ->
+    %% {Id, Total, Succeeded, Failed}
+    update_auth_attempt(Ip, Username, [{2, 1}, {4, 1}]).
+
+update_auth_attempt(Ip, Username, Incr) ->
+    %% It should default to false as per ip/user metrics could keep growing indefinitely
+    %% It's up to the operator to enable them, and reset it required
+    Id = case application:get_env(rabbit, return_per_user_auth_attempt_metrics) of
+             {ok, true} ->
+                 {Ip, Username};
+             {ok, false} ->
+                 {<<"">>, <<"">>}
```

Comment:
Could we not use any binaries in this case? Related to https://github.com/rabbitmq/rabbitmq-prometheus/pull/58#pullrequestreview-490798101